### ISSUE TITLE
WRONGTYPE error on v4 symbolication

### ISSUE THREAD

**Issue by peterbe** [mozilla-services/tecken#1022]
See https://sentry.prod.mozaws.net/operations/symbols-prod/issues/4609284/

```
ResponseError: WRONGTYPE Operation against a key holding the wrong kind of value
  File "django/core/handlers/exception.py", line 35, in inner
    response = get_response(request)
  File "django/core/handlers/base.py", line 128, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "django/core/handlers/base.py", line 126, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "newrelic/hooks/framework_django.py", line 544, in wrapper
    return wrapped(*args, **kwargs)
  File "tecken/base/decorators.py", line 165, in inner
    response = func(*args, **kwargs)
  File "django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "tecken/base/decorators.py", line 100, in wrapper
    return view_func(request, *args, **kwargs)
  File "markus/main.py", line 357, in _timer_decorator
    return fun(*args, **kwargs)
  File "tecken/symbolicate/views.py", line 843, in inner
    return view_function(request, json_body)
  File "tecken/symbolicate/views.py", line 920, in symbolicate_v4_json
    result = symbolicator.symbolicate(stacks, memory_map)
  File "tecken/symbolicate/views.py", line 165, in symbolicate
    informations = self.get_symbol_offsets(needed_modules_lookups)
  File "markus/main.py", line 357, in _timer_decorator
    return fun(*args, **kwargs)
  File "tecken/symbolicate/views.py", line 562, in get_symbol_offsets
    store.make_key(cache_key)
  File "newrelic/hooks/datastore_redis.py", line 77, in _nr_wrapper_Redis_method_
    result = wrapped(*args, **kwargs)
  File "redis/client.py", line 1985, in hlen
    return self.execute_command('HLEN', name)
  File "redis/client.py", line 668, in execute_command
    return self.parse_response(connection, command_name, **options)
  File "redis/client.py", line 680, in parse_response
    response = connection.read_response()
  File "redis/connection.py", line 629, in read_response
    raise response
```

**Comment by peterbe** [comment]
The way this can happen is that if a symbol didn't exist the first time. 
E.g. you include `mozglue.pdb/A9FF6A5A4E16883AFD16481F9298A75F1` in the memory map and the first time you query, it will put this into Redis like this:

```python
caches['store'].set('symbol:630cb:mozglue.pdb/A9FF6A5A4E16883AFD16481F9298A75F1', [], 60)
```
That's to store, in the redis store cache, that we tried to download it from S3 but failed so don't try again for 60 seconds (or whatever the TTL might be)

If you do try later, and now it exists in S3 you might, somehow, try to do an [`HLEN`](https://redis.io/commands/hlen) query on this and since it's not a hash map you can get that error. 

Working on making a reproducible unit test.

**Comment by peterbe** [comment]
There is no way to reproduce this other than manually delete a certain key that just can't happen in code. It's very likely this happened due to some strange race condition related to the fact that all the Redis store operations are not transactional. 

In particular, what happened was that for some reason we had *some* of the keys. When a symbol is requested in symbolication, we download it from S3 and store two things in Redis (simplified):

1. `cache.set('symbol:prefix:foo.pdb/HEXHEXHEX:keys', [123, 345])`
2. `cache.hmset('symbol:prefix:foo.pdb/HEXHEXHEX', {123: "A", 345:"B"})` (note the lack of `:keys` in the key)

But if a certain symbol is not found in S3, we just do:

1. `cache.set('symbol:prefix:foo.pdb/HEXHEXHEX', [])` (note that it's not a .hmset but a .set)

This is done to remember to not pester S3 for this if we've already concluded that it doesn't exist. Note! If you upload `foo.pdb/HEXHEXHEX/foo.sym` it will delete all and any of these keys.

What appears to have happened is that `symbol:prefix:foo.pdb/HEXHEXHEX:keys` existed but the hashmap `symbol:prefix:foo.pdb/HEXHEXHEX` did not. 

Because the fact that a symbol can't be found is stored for 60 * 60 seconds, we observe that the error in Sentry occurred for about a total of 1 hour. 

Most likely this error won't happen any time again soon. And if it does I reduced the cache-on-not-found-in-s3 from 1 hour to 1 minute just to make it less painful if this ever happens again.
